Fix related item review state 376

arsenico13 opened the pull request:
Fix the WorkflowException when requesting a fullobject with a related item that doesn't have a workflow (see issue #376).

I added one test about this topic. When tryint to run all `TestContentGet` I saw that all of them were skipped. I had to remove the check at the top of the test class:
```
if not HAS_AT:
    raise unittest.SkipTest("Skip tests if Archetypes is not present")
```

because this is a DX test.

For compatibilty reasons, do we need to add a HAS_DX check here?

mister-roboto commented:
@arsenico13 thanks for creating this Pull Request and help improve Plone!

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass.

Whenever you feel that the pull request is ready to be tested, either start all jenkins jobs pull requests by yourself, or simply add a comment in this pull request stating:

```
@jenkins-plone-org please run jobs
```

With this simple comment all the jobs will be started automatically.

Happy hacking!

arsenico13 commented:
@jenkins-plone-org please run jobs

tisto commented on CONTRIBUTORS.rst:
```diff
@@ -2,17 +2,20 @@
     place names and roles of the people who contribute to this package
     in this file, one to a line, like so:
 
-- Timo Stollenwerk, Original Author
```
@arsenico13 we usually just append names at the end as a best practice in the Plone community. I appreciate that you want to sort the authors by name but this never really worked. Most people add their names at the end, so I think we should keep things simple and just always append the names at the end...

arsenico13 replied:
Sorry, didn't know that!
Restored the old order and added missing contributors!

tisto commented on src/plone/restapi/tests/test_content_get.py:
```diff
@@ -23,8 +25,6 @@ class TestContentGet(unittest.TestCase):
     layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
 
     def setUp(self):
-        if not HAS_AT:
```
@arsenico13 does this test pass with AT?

arsenico13 replied:
Here the test is using the dexterity layer. The ones with AT are at the bottom of the file, line 181.

arsenico13 commented:
Hi @tisto, about your question:
> @arsenico13 does this test pass with AT?
Here the tests are using the dexterity layer. I removed those lines because they were actually preventig those tests from executing (the ones with AT are at the bottom of the file, line 181)
but now I added a new check and if dexterity is not present, the tests are skipped.
:)

arsenico13 commented:
I have another question.
All the tests run successfully but [here](https://jenkins.plone.org/job/pull-request-5.2/838/console) there is an error message in the console:

> ERROR: Step ‘Publish JUnit test result report’ failed: No test report files were found. Configuration error?

and I don't know what it is, sorry :(

arsenico13 commented:
@jenkins-plone-org please run jobs

tisto commented:
@arsenico13 this seems to be a Jenkins issue. We have troubles with Jenkins so I'd be ok to merge if Travis passes.

tisto commented:
@arsenico13 let's see if we can get a green Jenkins. If not I will merge and release over the weekend.
